NetTopologySuite/IO/ConfigurableBinaryReader.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using GeoAPI.IO;
+using NetTopologySuite.Utilities;
+
+namespace NetTopologySuite.IO
+{
+    /// <summary>
+    /// Extends the <see cref="BinaryReader" /> class to allow reading values in the specified format.    
+    /// </summary>
+    /// <remarks>
+    /// While <see cref="ConfigurableBinaryReader" /> extends <see cref="BinaryReader" /> 
+    /// adding methods for reading integer values (<see cref="ConfigurableBinaryReader.ReadInt32" />)
+    /// and double values (<see cref="ConfigurableBinaryReader.ReadDouble" />) in the specified format, 
+    /// this implementation overrides methods, such <see cref="BinaryReader.ReadInt32" /> 
+    /// and <see cref="BinaryReader.ReadDouble" /> and more, 
+    /// for reading values in the specified format.
+    /// </remarks>
+    public class ConfigurableBinaryReader : BinaryReader
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigurableBinaryReader"/> class.
+        /// </summary>
+        /// <param name="stream">The stream.</param>
+        public ConfigurableBinaryReader(Stream stream) : base(stream) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigurableBinaryReader"/> class.
+        /// </summary>
+        /// <param name="input">The supplied stream.</param>
+        /// <param name="encoding">The character encoding.</param>
+        /// <exception cref="T:System.ArgumentNullException">encoding is null. </exception>
+        /// <exception cref="T:System.ArgumentException">The stream does not support reading, the stream is null, or the stream is already closed. </exception>
+        public ConfigurableBinaryReader(Stream input, Encoding encoding) : base(input, encoding) { }
+
+        /// <summary>
+        /// Encoding type
+        /// </summary>
+        public ByteOrder EncodingType { get; set; }
```

Comment:
public ByteOrder `Endianess` { get; set; }

Reply:
The name came from `WKBWriter`. It has the `EncodingType` field. Change it anyway?

Reply:
I don't necessarily connect `EncodingType` to byte order or endianess.
I'd say either `Endianess` or `ByteOrder` is the right word.